src/main/java/org/prebid/server/bidder/bidscube/BidscubeBidder.java

```diff
@@ -0,0 +1,146 @@
+            throw new PreBidException("Missing required bidder parameters");
+        }
+
+        if (StringUtils.isEmpty(extImpBidscube.getPlacementId())) {
+            throw new PreBidException("Missing required bidder parameters");
+        }
+    }
+
+    private HttpRequest<BidRequest> createRequest(BidRequest request, Imp imp) {
+        return HttpRequest.<BidRequest>builder()
+                .method(HttpMethod.POST)
+                .uri(endpointUrl)
+                .headers(HttpUtil.headers())
+                .payload(request.toBuilder().imp(Collections.singletonList(imp)).build())
+                .body(mapper.encode(request))
+                .build();
+    }
+
+    @Override
+    public final Result<List<BidderBid>> makeBids(HttpCall<BidRequest> httpCall, BidRequest bidRequest) {
+        final List<BidderError> errors = new ArrayList<>();
+
+        try {
+            final BidResponse bidResponse = mapper.decodeValue(httpCall.getResponse().getBody(), BidResponse.class);
+            return Result.of(extractBids(bidResponse, errors), errors);
+        } catch (DecodeException | PreBidException e) {
+            return Result.withError(BidderError.badServerResponse(e.getMessage()));
+        }
+    }
+
+    private List<BidderBid> extractBids(BidResponse bidResponse, List<BidderError> errors) {
+        if (bidResponse == null || CollectionUtils.isEmpty(bidResponse.getSeatbid())) {
+            return Collections.emptyList();
+        }
+        return bidsFromResponse(bidResponse, errors);
+    }
+
+    private List<BidderBid> bidsFromResponse(BidResponse bidResponse, List<BidderError> errors) {
+        return bidResponse.getSeatbid().stream()
+                .filter(Objects::nonNull)
+                .map(SeatBid::getBid)
+                .filter(Objects::nonNull)
+                .flatMap(Collection::stream)
+                .map(bid -> constructBidderBid(bid, bidResponse, errors))
+                .filter(Objects::nonNull)
+                .collect(Collectors.toList());
+    }
+
+    private BidderBid constructBidderBid(Bid bid, BidResponse bidResponse, List<BidderError> errors) {
+        try {
+            final JsonNode bidExt = bid.getExt();
+            if (bidExt == null) {
+                throw new IllegalArgumentException();
+            }
+
+            final String bidTypeString = bidExt.path("prebid").path("type").asText("");
+            final BidType bidType = mapper.decodeValue(String.format("\"%s\"", bidTypeString), BidType.class);
+
+            return BidderBid.of(bid, resolveBidType(bidType), bidResponse.getCur());
+        } catch (IllegalArgumentException | DecodeException e) {
+            errors.add(BidderError.badInput("Unable to read bid.ext.prebid.type"));
+            return null;
+        }
+    }
+
+    private static BidType resolveBidType(BidType bidType) {
+        if (!POSSIBLE_BID_TYPES.contains(bidType)) {
+            return BidType.banner;
+        }
+        return bidType;
+    }
```

Comment:
Let's update this code to this one
```java
private static BidderBid constructBidderBid(Bid bid, BidResponse bidResponse, List<BidderError> errors) {
        final JsonNode bidExt = bid.getExt();
        final JsonNode prebidNode = isNotEmptyOrMissedNode(bidExt) ? bidExt.get("prebid") : null;
        final JsonNode typeNode = isNotEmptyOrMissedNode(prebidNode) ? prebidNode.get("type") : null;
        if (typeNode == null || !typeNode.isTextual()) {
            errors.add(BidderError.badInput("Unable to read bid.ext.prebid.type"));
            return null;
        }

        return BidderBid.of(bid, resolveBidType(typeNode.asText()), bidResponse.getCur());
    }

    private static boolean isNotEmptyOrMissedNode(JsonNode node) {
        return node != null && !node.isEmpty();
    }

    private static BidType resolveBidType(String bidType) {
        if (!POSSIBLE_BID_TYPES.contains(bidType)) {
            return BidType.banner;
        }
        return BidType.valueOf(bidType);
    }
```
And POSSIBLE_BID_TYPES to `ImmutableSet.of("banner", "video", "native")`
Now we can work only with String as in go and with unknown media types